web/src/components/IdeCascadeStudio/IdeCascadeStudio.js

```diff
@@ -0,0 +1,159 @@
+import { useMutation, useFlash } from '@redwoodjs/web'
+import { useAuth } from '@redwoodjs/auth'
+import { Link, routes, navigate } from '@redwoodjs/router'
+import { initialize } from 'src/cascade/js/MainPage/CascadeMain'
+import CascadeController from 'src/helpers/cascadeController'
+import IdeToolbar from 'src/components/IdeToolbar'
+import { useEffect, useState } from 'react'
+
+const DELETE_PART_MUTATION = gql`
+  mutation DeletePartMutation($id: Int!) {
+    deletePart(id: $id) {
+      id
+    }
+  }
+`
+const domNode = document.createElement('div').setAttribute('id', 'sickId')
+
+const IdeCascadeStudio = ({ part, saveCode, loading, error }) => {
+  const [code, setCode] = useState(part.code)
+  const { currentUser } = useAuth()
+  const canEdit = currentUser?.sub === part?.user?.id
+  useEffect(() => {
```

Comment:
Add a description to this useEffect, something along the lines of.

Cascade studio attatches to a div outside the react app, and so we are "opening" and "closing" it for the ide part of the app by displaying none or block. Which is why this useEffect returns a clean up function that hides the div again.